### PR TITLE
Wait for services to appear by default

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/core/task.toit [hunk 1]
```diff
@@ -175,8 +177,8 @@ class Task_ implements Task:
     if previous == this:
       // If we encounted a root-error, terminate the process.
       if exit_with_error_: __exit__ 1
-      // Check whether only background tasks are running.
-      if task_count_ == task_background_: __halt__
+      // Check whether no service definitions and only background tasks are running.
+      if ServiceManager_.is_empty and task_count_ == task_background_:  __halt__
```
Comment (floitsch) [inline lib/core/task.toit:181]:
```suggestion
      if ServiceManager_.is_empty and task_count_ == task_background_: __halt__
```

File: lib/system/services.toit [hunk 2]
```diff
@@ -323,6 +335,7 @@ class ServiceResourceProxyManager_ implements SystemMessageHandler_:
 
 class ServiceManager_ implements SystemMessageHandler_:
   static instance := ServiceManager_
+  static uninitialized/bool := true
```
Comment (floitsch) [inline lib/system/services.toit:338]: ?
```suggestion
  static initialized/bool := false
```

File: tools/snapshot.toit [hunk 3]
```diff
@@ -785,13 +785,13 @@ BYTE_CODES ::= [
   Bytecode "LOAD_LITERAL_WIDE"          3 OP_SL "load literal wide",
   Bytecode "LOAD_NULL"                  1 OP "load null",
   Bytecode "LOAD_SMI_0"                 1 OP "load smi 0",
-  Bytecode "LOAD_8_SMI_0"               1 OP "load 8 smi 0",
+  Bytecode "LOAD_SMIS_0"                2 OP_BU "load smis 0",
```
Comment (floitsch) [inline tools/snapshot.toit:788]: I would prefer these in a separate PR as well.

**Review by floitsch — APPROVED** [review]
LGTM.

File: lib/system/services.toit (hunk 2 above)
Comment (kasperl) [reply lib/system/services.toit:338]: I think I can avoid the whole thing in a follow-up. It just requires some surgery on the rpc test.

File: tests/tls_test_slow.toit [hunk 4]
```diff
@@ -55,7 +55,7 @@ run_tests:
     // "ecc384.badssl.com",
     "rsa2048.badssl.com",
     "rsa4096.badssl.com",
-    "extended-validation.badssl.com",
+    // "extended-validation.badssl.com",
```
[the remark this answers is not in the document]
Comment (kasperl) [reply tests/tls_test_slow.toit:58]: That's a valid point. I'll try to do that next time :)

**Pull request merged**